lib/DataBuilder/ProjectDataBuilder.php

```diff
@@ -200,8 +200,10 @@ private function isProjectVersionEquals(array $a, array $b): bool
         if ($a['name'] === $b['name']) {
             return true;
         }
-
-        return $a['branchName'] === $b['branchName'];
+        if (array_key_exists('branchName', $a) && array_key_exists('branchName', $b)) {
+            return $a['branchName'] === $b['branchName'];
+        }
+        return false;
```

Comment:
```suggestion
        if (! isset($b['branchName'])) {
            return false;
        }

        return $a['branchName'] === ($b['branchName']);
```

`$a` contains the data from the repository and will have the `branchName`.